functions.php

```diff
@@ -281,4 +281,131 @@ function add_queries_tags_categories($wp_query) {
+
+    // Get the query & post information
+    global $post,$wp_query;
+
+    // Do not display on the homepage
+    if ( !is_front_page() ) {
+
+        // Build the breadcrums
+        echo '<ul class="' . $breadcrumbs_class . '">';
+
+        if ( is_archive() && !is_tax() && !is_category() && !is_tag() ) {
+
+            echo '<li class="item-current item-archive"><strong class="bread-current bread-archive">' . post_type_archive_title($prefix, false) . '</strong></li>';
+
+        } else if ( is_archive() && is_tax() && !is_category() && !is_tag() ) {
+
+            // If post is a custom post type
+            $post_type = get_post_type();
+
+            // If it is a custom post type display name and link
+            if($post_type != 'post') {
+
+                $post_type_object = get_post_type_object($post_type);
+                $post_type_archive = get_post_type_archive_link($post_type);
+
+                echo '<li class="item-cat item-custom-post-type-' . $post_type . '"><a class="bread-cat bread-custom-post-type-' . $post_type . '" href="' . $post_type_archive . '" title="' . $post_type_object->labels->name . '">' . $post_type_object->labels->name . '</a></li>';
+                echo '<li class="separator"> ' . $separator . ' </li>';
+
+            }
+
+        } else if ( is_single() ) {
+
+            // If post is a custom post type
+            $post_type = get_post_type();
+
+            // If it is a custom post type display name and link
+            if($post_type != 'post') {
+
+                $post_type_object = get_post_type_object($post_type);
+                $post_type_archive = get_post_type_archive_link($post_type);
+
+                echo '<li class="item-cat item-custom-post-type-' . $post_type . '"><a class="bread-cat bread-custom-post-type-' . $post_type . '" href="' . $post_type_archive . '" title="' . $post_type_object->labels->name . '">' . $post_type_object->labels->name . '</a></li>';
+                echo '<li class="separator"> ' . $separator . ' </li>';
+
+            }
+
+            // Get post category info
+            $category = get_the_category();
+
+            if(!empty($category)) {
+
+                // Get last category post is in
+                $last_category = end(array_values($category));
+
+                // Get parent any categories and create array
+                $get_cat_parents = rtrim(get_category_parents($last_category->term_id, true, ','),',');
+                $cat_parents = explode(',',$get_cat_parents);
+
+                // Loop through parent categories and store in variable $cat_display
+                $cat_display = '';
+                foreach($cat_parents as $parents) {
+                    $cat_display .= '<li class="item-cat">'.$parents.'</li>';
+                    $cat_display .= '<li class="separator"> ' . $separator . ' </li>';
+                }
+
+            }
+
+            // Check if the post is in a category
+            if(!empty($last_category)) {
```

Comment:
It looks like this  `if` block can be merged into the previous `if(!empty($category))` block.

In this particular case`$last_category` and `$category` are functionally the same - $last_category will be non-empty if $category is non-empty.